particle-iot/device-os
# [Gen2] Add cellular registration intervention commands to Gen2

### Problem

Add cellular AT intervention commands to Gen2, to improve/speed up registration process. This is particularly helpful in scenarios where the radio struggles to connect to an invalid operator.

### Solution

1. Added `CellularRegistrationStatusMdm` similar to the one in Gen3 to run intervention commands
2. Replaced the `csd, psd, and eps` members of `NetStatus` with `CellularRegistrationStatusMdm`'s status methods for consistency.

### Steps to Test

Run any tinker-like app on Gen3 which enables connecting to the network. Try removing the antenna or using an invalid SIM, even better if there are known network issues in a given area to check how the the intervention commands run and help register to a network.

### Example App

```c
void setup() {
}

void loop() {
}
```



### References

- [ch69276](https://app.clubhouse.io/particle/story/69276/port-gen-3-cellular-registration-intervention-logic-to-gen2)

---

### Completeness
